spec/support/pages/projects/settings/life_cycle.rb

```diff
@@ -0,0 +1,100 @@
+# -- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2010-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+# ++
+
+require "support/pages/page"
+
+module Pages
+  module Projects
+    module Settings
+      class LifeCycle < Pages::Page
+        attr_accessor :project
+
+        def initialize(project)
+          super()
+
+          self.project = project
+        end
+
+        def path
+          "/projects/#{project.identifier}/settings/life_cycle"
+        end
+
+        # Checks if the life cycle steps are listed in the order given and with the correct toggle state.
+        # @param life_cycle_definitions [Hash{LifeCycleElement => Boolean}]
+        def expect_listed(**life_cycle_definitions)
```

Comment:
As this check is used to validate the result of filter, I think it is better to find all row elements and check that they match the order and active status of passed definitions